subsys/sal/sid_pal/src/CMakeLists.txt

```
@@ -63,7 +63,11 @@ zephyr_library_sources_ifdef(CONFIG_SIDEWALK_SW_INTERRUPTS sid_sw_interrupts.c)
 
 zephyr_library_sources_ifdef(CONFIG_SIDEWALK_DELAY sid_delay.c)
 
+if(NOT (CONFIG_SOC_NRF52840 AND CONFIG_SIDEWALK_NRF52840_NRFX_SPI))
```

minor: It would be easer to tread when the if statement is not a negation
```suggestion
if((CONFIG_SOC_NRF52840 AND CONFIG_SIDEWALK_NRF52840_NRFX_SPI))
zephyr_library_sources_ifdef(CONFIG_SIDEWALK_SPI_BUS sid_pal_serial_bus_nrfx_spi.c)
else()
zephyr_library_sources_ifdef(CONFIG_SIDEWALK_SPI_BUS sid_pal_serial_bus_spi.c)
endif() # CONFIG_SOC_NRF52840 AND CONFIG_NRF52840_NRFX_SPI
```

---

Done